torch/distributed/nn/jit/templates/remote_module_template.py

```diff
@@ -1,19 +1,63 @@
 #!/usr/bin/python3
 
-REMOTE_MODULE_TEMPLATE = """from typing import *
+
+def get_remote_module_template(enable_moving_cpu_tensors_to_cuda: bool):
+    return _TEMPLATE_PREFIX + (
+        _REMOTE_FORWARD_TEMPLATE_ENABLE_MOVING_CPU_TENSORS_TO_CUDA
+        if enable_moving_cpu_tensors_to_cuda
+        else _REMOTE_FORWARD_TEMPLATE
+    )
+
+
+_TEMPLATE_PREFIX = """from typing import *
 
 import torch
 import torch.distributed.rpc as rpc
 from torch import Tensor
 from torch._jit_internal import Future
 from torch.distributed.rpc import RRef
-from typing import Tuple
+from typing import Tuple  # pyre-ignore: unused import
 
 
 {assign_module_interface_cls}
 
 
+def forward_async(self, {arg_types}){arrow_and_future_return_type}:
+    args = (self.module_rref, self.device, self.is_device_map_set, {args})
+    kwargs = {{{kwargs}}}
+    return rpc.rpc_async(
+        self.module_rref.owner(),
+        _remote_forward,
+        args,
+        kwargs,
+    )
+
+
+def forward(self, {arg_types}){arrow_and_return_type}:
+    args = (self.module_rref, self.device, self.is_device_map_set, {args})
+    kwargs = {{{kwargs}}}
+    ret_fut = rpc.rpc_async(
+        self.module_rref.owner(),
+        _remote_forward,
+        args,
+        kwargs,
+    )
+    return ret_fut.wait()
+
+
+_generated_methods = [
+    forward_async,
+    forward,
+]
+
+
 {jit_script_decorator}
+"""
+
+# This template may cause typing error (the mismatch between ``Tuple[()]`` and ``Tuple[Any]``)
```

Comment:
can you add a TODO here to make sure we merge these two template together in the future when TorchScript syntax improves?